### PR TITLE
[4.0] Associations tooltip id accessibility

### PR DESCRIPTION
**Pull request by brianteeman**
On a multilingual site with more than two languages the language indicators have a div for displaying the _tooltip_

The way the id for the div was generated it might not be unique. This PR ensures that it is unique.

Applied to menus, categories, content, contacts, newsfeeds (don't think there is anywhere else?)

image

### CODE REVIEW

File: administrator/components/com_content/Service/HTML/AdministratorService.php [hunk 1]
```diff
@@ -101,7 +101,6 @@ public function association($articleid)
 
 			$html = LayoutHelper::render('joomla.content.associations', $items);
 		}
-
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/AdministratorService.php]: Revert

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d5faf059de09ad6a2bfa73ea136b5839996461c2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27370">issues.joomla.org/tracker/joomla-cms/27370</a>.</sub>

**Comment by Quy** [comment]
Please confirm that the title attribute should be removed since there is a tooltip for it. I will do a separate a PR.

**Comment by astridx** [comment]
If tested this successful.

I did the following:

1. I had checked the background for the Post Installation Hint
Control Panel   Test   Administration
 It was

`#f9acaa;`

```
.badge-danger {
    color: 
#3b0d0c;
background-color:
#f9acaa;
border: 1px solid
    #ba2926;
}
```
2. I run `git fetch origin pull/27357/head:qicon` and changed to `qicon` branch.

3. I run `npm i`

4. After reloading the site I did not realize a change. But in CSS I see the new background color.
`#f5b9b7;`
```
.badge-danger {
    color: 
#3b0d0c;
background-color:
#f5b9b7;
border: 1px solid
    #ba2926;
}
```

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on ab43a1ee84e307bc14d04d3fa2cd89d63862b29f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27370">issues.joomla.org/tracker/joomla-cms/27370</a>.</sub>

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on ab43a1ee84e307bc14d04d3fa2cd89d63862b29f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27370">issues.joomla.org/tracker/joomla-cms/27370</a>.</sub>

File: administrator/components/com_categories/Service/HTML/AdministratorService.php [hunk 2]
```diff
@@ -86,7 +86,7 @@ public function association($catid, $extension = 'com_content')
 						$classes  = 'badge badge-secondary';
 
 						$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
-							. '<div role="tooltip" id="tip' . (int) $item->id . '">' . $tooltip . '</div>';
+							. '<div role="tooltip" id="tip' . (int) $catid . (int) $item->id . '">' . $tooltip . '</div>';
```
Comment (HLeithner) [inline administrator/components/com_categories/Service/HTML/AdministratorService.php]: It's possible that this is not a unique string, you have to add a separator between the numbers like a dash.
```suggestion
							. '<div role="tooltip" id="tip-' . (int) $catid . '-' . (int) $item->id . '">' . $tooltip . '</div>';
```

File: administrator/components/com_contact/Service/HTML/AdministratorService.php [hunk 3]
```diff
@@ -96,7 +96,7 @@ public function association($contactid)
 						$classes = 'badge badge-secondary';
 
 						$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
-							. '<div role="tooltip" id="tip' . (int) $item->id . '">' . $tooltip . '</div>';
+							. '<div role="tooltip" id="tip' . (int) $contactid . (int) $item->id . '">' . $tooltip . '</div>';
```
Comment (HLeithner) [inline administrator/components/com_contact/Service/HTML/AdministratorService.php]:
```suggestion
							. '<div role="tooltip" id="tip-' . (int) $contactid . '-' . (int) $item->id . '">' . $tooltip . '</div>';
```

File: administrator/components/com_content/Service/HTML/AdministratorService.php [hunk 4]
```diff
@@ -89,7 +89,7 @@ public function association($articleid)
 						$classes = 'badge badge-secondary';
 
 						$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
-							. '<div role="tooltip" id="tip' . (int) $item->id . '">' . $tooltip . '</div>';
+							. '<div role="tooltip" id="tip' . (int) $articleid . (int) $item->id . '">' . $tooltip . '</div>';
```
Comment (HLeithner) [inline administrator/components/com_content/Service/HTML/AdministratorService.php]:
```suggestion
							. '<div role="tooltip" id="tip-' . (int) $articleid . '-' . (int) $item->id . '">' . $tooltip . '</div>';
```

File: administrator/components/com_menus/Service/HTML/Menus.php [hunk 5]
```diff
@@ -96,7 +96,7 @@ public function association($itemid)
 						$classes = 'badge badge-secondary';
 
 						$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
-							. '<div role="tooltip" id="tip' . (int) $item->id . '">' . $tooltip . '</div>';
+							. '<div role="tooltip" id="tip' . (int) $itemid . (int) $item->id . '">' . $tooltip . '</div>';
```
Comment (HLeithner) [inline administrator/components/com_menus/Service/HTML/Menus.php]:
```suggestion
							. '<div role="tooltip" id="tip-' . (int) $itemid . '-' . (int) $item->id . '">' . $tooltip . '</div>';
```

File: administrator/components/com_newsfeeds/Service/HTML/AdministratorService.php [hunk 6]
```diff
@@ -87,7 +87,7 @@ public function association($newsfeedid)
 						$classes = 'badge badge-secondary';
 
 						$item->link = '<a href="' . $url . '" title="' . $item->language_title . '" class="' . $classes . '">' . $text . '</a>'
-							. '<div role="tooltip" id="tip' . (int) $item->id . '">' . $tooltip . '</div>';
+							. '<div role="tooltip" id="tip' . (int) $newsfeedid . (int) $item->id . '">' . $tooltip . '</div>';
```
Comment (HLeithner) [inline administrator/components/com_newsfeeds/Service/HTML/AdministratorService.php]:
```suggestion
							. '<div role="tooltip" id="tip-' . (int) $newsfeedid . '-' . (int) $item->id . '">' . $tooltip . '</div>';
```

**Review by HLeithner — CHANGES_REQUESTED** [review]
In you case it possible to have cat id 1 and language 11 which would be tip111 but it's possible that you have cat id 11 and language 1 which would be 111 as well. Adding dashes changes this to tip-1-11 or tip-11-1

**Comment by astridx** [comment]
I did the following:

1. I run `git fetch origin pull/27370/head:associationdiv` 
2. After that I installed French and German language and activated the `language filter` plugin.
3. I created a French a German and an English article and set up the associations.
4. I debug and checked the id [here](https://github.com/joomla/joomla-cms/blob/f51e6a73790221656815d8f937492bb536a9d768/administrator/components/com_content/Service/HTML/AdministratorService.php#L91).
5. I checked out the branch `associationdiv`
6. I run debugger again an see, that the ID [here](https://github.com/joomla/joomla-cms/blob/f51e6a73790221656815d8f937492bb536a9d768/administrator/components/com_content/Service/HTML/AdministratorService.php#L91) was correctly changed.

**Comment by brianteeman** [comment]
Good spot @HLeithner I have applied those changes

**Comment by HLeithner** [comment]
@brianteeman you missed a button^^

**Comment by brianteeman** [comment]
@HLeithner I was on my phone :(

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6d21ea210b125660db3963735c760a947875d279<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27370">issues.joomla.org/tracker/joomla-cms/27370</a>.</sub>

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on 6d21ea210b125660db3963735c760a947875d279<br /><br />I did 'git pull origin pull/27370/head:associationdiv' and repeated my test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27370">issues.joomla.org/tracker/joomla-cms/27370</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**